src/Orchard.Web/Modules/Orchard.Autoroute/Drivers/AutoroutePartDriver.cs

```diff
@@ -20,54 +25,99 @@ public class AutoroutePartDriver : ContentPartDriver<AutoroutePart> {
         private readonly IAutorouteService _autorouteService;
         private readonly IAuthorizer _authorizer;
         private readonly INotifier _notifier;
+        private readonly ICultureManager _cultureManager;
+        private readonly IHttpContextAccessor _httpContextAccessor;
 
         public AutoroutePartDriver(
-            IAliasService aliasService, 
+            IAliasService aliasService,
             IContentManager contentManager,
             IAutorouteService autorouteService,
             IAuthorizer authorizer,
-            INotifier notifier) {
+            INotifier notifier,
+            ICultureManager cultureManager,
+            IHttpContextAccessor httpContextAccessor) {
             _aliasService = aliasService;
             _contentManager = contentManager;
             _autorouteService = autorouteService;
             _authorizer = authorizer;
             _notifier = notifier;
+            _cultureManager = cultureManager;
+            _httpContextAccessor = httpContextAccessor;
 
             T = NullLocalizer.Instance;
         }
 
         public Localizer T { get; set; }
 
-        protected override string Prefix { get { return "Autoroute"; }}
-
         protected override DriverResult Editor(AutoroutePart part, dynamic shapeHelper) {
             return Editor(part, null, shapeHelper);
         }
 
         protected override DriverResult Editor(AutoroutePart part, IUpdateModel updater, dynamic shapeHelper) {
-
             var settings = part.TypePartDefinition.Settings.GetModel<AutorouteSettings>();
-            
+            var itemCulture = _cultureManager.GetSiteCulture();
+
+            //if we are editing an existing content item
+            if (part.Record.Id != 0) {
+                ContentItem contentItem = _contentManager.Get(part.Record.ContentItemRecord.Id);
+                var aspect = contentItem.As<ILocalizableAspect>();
+
+                if (aspect != null) {
+                    itemCulture = aspect.Culture;
+                }
+            }
+
+            if (settings.UseCulturePattern) {
+                //if we are creating from a form post we use the form value for culture
+                HttpContextBase context = _httpContextAccessor.Current();
+                if (!String.IsNullOrEmpty(context.Request.Form["Localization.SelectedCulture"])) {

```

Comment:
Since the LocalizatioPart is not already attached to the content type I found no other way to get this value.